src/java.base/share/classes/java/lang/ModuleLayer.java

```diff
@@ -297,6 +299,32 @@ public Controller addOpens(Module source, String pn, Module target) {
             source.implAddOpens(pn, target);
             return this;
         }
+
+        /***
+         * Enables <a href="foreign/package-summary.html#restricted">native access</a>
+         * for a module in the layer if the caller module already has native access.
+         *
+         * <p> This method is <a href="foreign/package-summary.html#restricted"><em>restricted</em></a>.
+         * Restricted methods are unsafe, and, if used incorrectly, their use might crash
+         * the JVM or, worse, silently result in memory corruption. Thus, clients should refrain
+         * from depending on restricted methods, and use safe and supported functionalities,
+         * where possible.
+         *
+         * @param target the module that's given native access
+         *
+         * @return This controller
+         *
+         * @throws IllegalCallerException If the caller module doesn't have native access.
```

Comment:
This is a restricted method, and needs the verbiage javadoc around restricted method. See e.g. `Linker::nativeLinker`. It should also probably specify what happens when called with no module in the call stack (possible if call comes from JNI). See `SymbolLookup::libraryLookup` for some hints. (note - the code already does the correct thing in Reflection.java)